variables.tf

```diff
@@ -1,3 +1,9 @@
+variable "admin_username" {
+  description = "(Required) The admin username of the VM that will be deployed."
```

Comment:
I like the idea that we should add `nullable = false` here, but should we keep `default = azureuser` for this variable in case the user doesn't care about the username? If we do that, we might also need to remove `(Required)` from the description.

Reply:
By doing that we mimic, more or less, what the Portal does (it suggests 'azureuser' automatically as the Username) making it more user friendly, but then shall we provide `var.admin_username` (`local.virtual_machine.admin_username`?) as an output? Final users may not know that this module defaults to 'azureuser'.

Reply:
Good idea! Please do output actual `admin_username` for this module, after that we're good to run the e2e test.